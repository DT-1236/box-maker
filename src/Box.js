import React, { Component } from 'react';

class Box extends Component {
  constructor(props) {
    super(props);
    this.handleRemove = this.handleRemove.bind(this);
  }

  handleRemove() {
    this.props.remove(this.props.id);
  }

  render() {
    return (
      <div>
        <div className="Box" style={this.props.style} />
        <button className="delete-button" onClick={this.handleRemove}>
          x
        </button>
      </div>
    );
  }
}

export default Box;
