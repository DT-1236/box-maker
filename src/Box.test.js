import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Box from './Box';

describe('smoke tests for mount/shallow', () => {
  it('renders without crashing', () => {
    shallow(<Box />);
  });

  it('mounts without crashing', () => {
    mount(<Box />);
  });
});

describe('snapshot testing', () => {
  it('passes simple snapshots', () => {
    expect(toJson(shallow(<Box />))).toMatchSnapshot();
    expect(toJson(mount(<Box />))).toMatchSnapshot();
  });

  it('passes snapshots with styling', () => {
    const style = { backgroundColor: '#444444', height: '25px', width: '50px' };
    const wrapper = mount(<Box style={style} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('gives proper style attributes to the rendered Box div and matches the snapshot', () => {
    const style = { backgroundColor: '#444444', height: '25px', width: '50px' };
    const wrapper = mount(<Box style={style} />);
    const mountedBoxStyle = wrapper.find('.Box').props().style;
    expect(mountedBoxStyle).toHaveProperty('height', '25px');
    expect(mountedBoxStyle).toHaveProperty('width', '50px');
    expect(mountedBoxStyle).toHaveProperty('backgroundColor', '#444444');
  });
});
