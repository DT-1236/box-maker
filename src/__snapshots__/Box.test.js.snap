// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing passes simple snapshots 1`] = `
<div>
  <div
    className="Box"
  />
  <button
    className="delete-button"
    onClick={[Function]}
  >
    x
  </button>
</div>
`;

exports[`snapshot testing passes simple snapshots 2`] = `
<Box>
  <div>
    <div
      className="Box"
    />
    <button
      className="delete-button"
      onClick={[Function]}
    >
      x
    </button>
  </div>
</Box>
`;

exports[`snapshot testing passes snapshots with styling 1`] = `
<Box
  style={
    Object {
      "backgroundColor": "#444444",
      "height": "25px",
      "width": "50px",
    }
  }
>
  <div>
    <div
      className="Box"
      style={
        Object {
          "backgroundColor": "#444444",
          "height": "25px",
          "width": "50px",
        }
      }
    />
    <button
      className="delete-button"
      onClick={[Function]}
    >
      x
    </button>
  </div>
</Box>
`;
